src/Lynx/Expr.purs

```diff
@@ -592,6 +597,11 @@ toExprA = case _ of
   Equal x y -> EqualA (toExprA x) (toExprA y) (reflectIn x) (reflectOut x)
   Print x -> PrintA (toExprA x) (reflectIn x) "String"
   Lookup x y -> LookupA x (toExprA y) (reflectOut y)
+  Collection x ->
+    CollectionA
+      (map toExprA x)
+      (maybe "TODO" reflectIn $ Data.Array.head x)
```

Comment:
How do we deal with the types here? Does it matter? Maybe we need to require at least one element? Maybe we don't need to talk about individual types too deeply? Maybe we're thinking at the wrong level?

Reply:
Do you mean the type of `x`?

Reply:
I mean the whole collection. The `ExprA` data type is supposed to be `Expr` annotated with the input and output types of the expression. E.g. the input type of `Equal` is whatever the input types of the sub `Expr`s are, and the output type of `Equal` is always `Boolean`.

For the idea of collections, what is its input type, and what is its output type? I would think we're dealing with homogeneous collections of things, so the input and output type would be related to the items in the collection. But if we don't have any element, where do we get that type information?

Reply:
Is `Collection` in the right place? That is, should it be a `Expr` as opposed to an `ExprType`? Which, we already have `Array` for. It's been a while since I've really dealt with this module, but I'm having trouble figuring out what we'd do with this `Collection`.

Reply:
`Expr` is used to define things in `Lynx.Form`. `ExprType` is the evaluated value of the `Expr`. If we put `Collection` in `ExprType`, we cannot evaluate it (it's already evaluated) and it effectively becomes static. I assumed we wanted collections of expressions to be dynamic. Is that not correct?

Looking at what we do for `Array`, we say the type is `"Array"`, we don't look underneath at any elements. It seems like that's what we should also do for `Collection`.

Reply:
> If we put `Collection` in `ExprType`, we cannot evaluate it (it's already evaluated) and it effectively becomes static. I assumed we wanted collections of expressions to be dynamic. Is that not correct?

My understanding of why we need `Collection`, is for the values of "Array" input types. Right now values are always static. I'm not sure what it would mean if a user could set the value of a field to an expression.

I.e., right now [`Lynx.Expr.Form.setValue` takes `InputSource ExprType` as an argument](https://github.com/citizennet/purescript-lynx/blob/master/src/Lynx/Form.purs#L456). If we were to take `InputSource Expr` instead, we'd have to allow them to build expressions in our UI. Something we will need to figure out for the form builder, actually, but I don't think that's something we are trying to support for regular Lynx forms.

Reply:
Is there anything to do on the `Expr`/`ExprType` side? If we don't need to have arrays of things at the `Expr` level, it seems like there's no change required. I'm more than happy to close this PR.

Reply:
Hmm, yeah I'm not sure there is. I say perhaps we start with adding arrays/collections to our layout types and see how far we can get with that.

Reply:
Sounds good. Thanks for the review you two!